### DISCUSSION TITLE
edgemesh service discovery problem

### DISCUSSION THREAD

**Discussion by liaohw123** [kubeedge/kubeedge#2942] [General]
In kubeedge version 1.6, edgemesh is used for service discovery. Sometimes the load balancing address of the go chassis framework < hostip: hostport > is not the address of the pod to the current service, which leads to access failure. In the source code, it is found that the pods in **FindMicroServiceInstances** method,   **pods, err: = ESD. Getpods (name, namespace)** get all the pods, Not only the pod corresponding to the current service?

**Comment by ZBoIsHere** [comment]
@liaohw123 Hello, edgemesh now is decoupling from edgecore, you can get the latest code from https://github.com/kubeedge/edgemesh, the new edgemesh get crd like pod from edgecore using list-watch but not mqtt. welcome to use the latest edgemesh. if you encounter any problem, you can raise issue on edgemesh repository.
